omero/sysadmins/unix/install-web/walkthrough/omeroweb-install-debian10-ice3.6.rst

```diff
@@ -0,0 +1,259 @@
+
+
+
+Configuring OMERO.web
+---------------------
+
+**The following steps are run as the omero-web system user.**
+
+For convenience the main OMERO.web configuration options have been defined as environment variables. You can either use your own values, or alternatively use the following ones::
+
+    export OMERODIR=/opt/omero/web/omero-web
+    export WEBPORT=80
+    export WEBSERVER_NAME=localhost
+
+
+Configure OMERO.web and create the NGINX OMERO configuration file::
+
+    export PATH=/opt/omero/web/venv3/bin:$PATH
+
+
+    omero config set omero.web.application_server wsgi-tcp
+    omero web config nginx --http "${WEBPORT}" --servername "${WEBSERVER_NAME}" > /opt/omero/web/omero-web/nginx.conf.tmp
+
+For more customization, please read :ref:`customizing_your_omero_web_installation`.
+
+
+Configuring Gunicorn
+--------------------
+
+**The following steps are run as the omero-web system user.**
+
+Additional settings can be configured by changing the following properties:
+
+    - :property:`omero.web.application_server.max_requests` to 500
+
+    - :property:`omero.web.wsgi_workers` to (2 x NUM_CORES) + 1
+
+      .. note::
+          **Do not** scale the number of workers to the number of clients
+          you expect to have. OMERO.web should only need 4-12 worker
+          processes to handle many requests per second.
+
+    - :property:`omero.web.wsgi_args` Additional arguments. For more details
+      check `Gunicorn Documentation <https://docs.gunicorn.org/en/stable/settings.html>`_.
+
+
+
+Configuring NGINX
+-----------------
+
+**The following steps are run as root.**
+
+Copy the generated configuration file into the NGINX configuration directory, disable the default configuration and start NGINX::
+
+    sed -i.bak -re 's/( default_server.*)/; #\1/' /etc/nginx/nginx.conf
+    rm /etc/nginx/sites-enabled/default
+    cp /opt/omero/web/omero-web/nginx.conf.tmp /etc/nginx/conf.d/omeroweb.conf
+
+    service nginx start
+
+
+Running OMERO.web
+-----------------
+
+**The following steps are run as root.**
+
+Install `WhiteNoise <http://whitenoise.evans.io/>`_::
+
+    /opt/omero/web/venv3/bin/pip install --upgrade 'whitenoise<4'
+
+
+**The following steps are run as the omero-web system user.**
+
+
+Configure WhiteNoise and start OMERO.web manually to test the installation::
+
+    omero config append -- omero.web.middleware '{"index": 0, "class": "whitenoise.middleware.WhiteNoiseMiddleware"}'
+
+    omero web start
+
+    # Test installation e.g. curl -sL localhost:4080
+
+    omero web stop
+
+
+Automatically running OMERO.web
+-------------------------------
+
+
+**The following steps are run as root.**
+
+Should you wish to run OMERO.web automatically, a `init.d` file could be created. See below an example file `omero-web-init.d`::
```

Comment:
Should probably be migrated to be a service file but of course that can be a todo; https://linux.slashdot.org/story/19/12/28/1555250/debian-project-votes-systemd-but-we-support-exploring-alternatives